### ISSUE TITLE
Moving Resource Hints

### ISSUE THREAD

**Issue by plehegar** [w3c/whatwg-coord#6]
- [x] Waiting on Web Perf WG to publish a CR?
- [ ] WHATWG published as part of HTML
- [x] W3C retires the WD or CR and do the /TR redirect to /TR/html

**Comment by plehegar** [comment]
[Web Performance June 24 minutes](https://docs.google.com/document/d/1P_Of7nCFf_UkE_Bvz6rFCNcETPZ_IyxQqRcUsx4L6do/edit#heading=h.3b3idaeuca8x)

**Comment by caribouW3** [comment]
https://www.w3.org/TR/2023/DISC-resource-hints-20230314/Overview.html    published

**Comment by annevk** [comment]
I hope you don't me reopening this, but I think there's a tiny bit of work left here still.

@noamr could you confirm the WHATWG published this as part of HTML?

As I understand it we define `dns-prefetch`, `preconnect`, and `prefetch`. And `prerender` is "disappeared" due to lack of multiple implementations. Is that correct?

Would you be willing to update the Web Linking registrations in https://github.com/protocol-registries/link-relations to account for this? I'm not entirely sure what to do about `prerender`, I guess don't update the reference, but do add a note saying it's essentially unused?

**Comment by noamr** [comment]
We still need to define `dns-prefetch` in detail, but what we have in HTML is equivalent to what we had in resource hints so keeping resource hints doesn't add anything.




 

> I hope you don't me reopening this, but I think there's a tiny bit of work left here still.
> 
> @noamr could you confirm the WHATWG published this as part of HTML?
> 
> As I understand it we define `dns-prefetch`, `preconnect`, and `prefetch`. And `prerender` is "disappeared" due to lack of multiple implementations. Is that correct?
> 
> Would you be willing to update the Web Linking registrations in https://github.com/protocol-registries/link-relations to account for this? I'm not entirely sure what to do about `prerender`, I guess don't update the reference, but do add a note saying it's essentially unused?

We still need to define `dns-prefetch` in detail, but what we have in HTML is equivalent to what we had in resource hints so keeping resource hints doesn't add anything. I confirm that HTML has all the details that resource hints had before.

I'll happily follow up to update the registry. I think we should keep prerender there, it's still implemented inside chrome. Does the IANA registration require multiple implementations?

**Comment by caribouW3** [comment]
@annevk I'm not archiving the resource-hints repository yet then.

**Comment by annevk** [comment]
@noamr it doesn't and I think you cannot remove registered items. I was thinking we would add a note to `prerender` (autocorrect keeps making that pretender, which is kinda funny) that it's not standardized. https://github.com/whatwg/html/issues/9015 suffices for `dns-prefetch` follow-up I think.

@caribouW3 I don't foresee any further changes to Resource Hints to be necessary, if that helps.

**Comment by domenic** [comment]
IIUC, the IANA registrations are for `Link` headers, but only `preconnect` (and `preload`) is supported in `Link` headers. `prefetch`, `dns-prefetch`, and `prerender` are not, so they should be de-registered if possible. If that's not possible for some reason, that seems bad and like a bug that IANA needs to address?

Note also the request in https://github.com/w3c/resource-hints/issues/100#issuecomment-1469114376 to make this move like other W3C -> WHATWG moves, and not publish a draft containing out of date normative content. (Even if it's marked as discontinued.)

**Comment by sideshowbarker** [comment]
https://www.w3.org/TR/resource-hints/ now already redirects to https://html.spec.whatwg.org/#linkTypes

https://www.w3.org/TR/2023/DISC-resource-hints-20230314/ has already been published. But Domenic’s comment at https://github.com/w3c/whatwg-coord/issues/6#issuecomment-1469119541 suggests that maybe we don’t actually want hanging around at all. Can we completely remove it?

**Comment by domenic** [comment]
Oh, if /TR redirects, then having a dated URL like that with content seems fine. (At least, in https://wiki.whatwg.org/wiki/Fork_tracking, I've only been tracking TR and ED URLs. So at this point Resource Hints is at least as good as all those other forks.)

So I think the only remaining issues here are the IANA-related ones?

**Comment by sideshowbarker** [comment]
> So I think the only remaining issues here are the IANA-related ones?

Sounds like. But then, is this the right place to be tracking those, or should we be using https://github.com/whatwg/html/issues tracker?

**Comment by domenic** [comment]
I'd be happy to use the HTML issue tracker, although maybe we should let @annevk chime in.

**Comment by annevk** [comment]
One reason we tracked that here as well before is because of the ownership change and the IANA maintainers might want a place to ask questions. Might be fine either way though.

**Comment by sideshowbarker** [comment]
I would prefer this be moved to the HTML issue tracker.

If the IANA maintainers end up wanting to ask questions about the ownership change — and there’s some need for somebody from the W3C to respond — then I think I or PLH or somebody could just as easily respond to that in an HTML issue-tracker issue as we could here. Or else we could at that time re-open this issue. Or something.

Issues in this repo have relatively very low visibility. And in practice this issue tracker exists to track issues that both don’t fit into any other existing issue tracker and moreover, that require some action _from the W3C_ to resolve: Most commonly, the need to deal with standards previously published in TR that nobody should be using any longer because they’ve been superseded by WHATWG standards.

So unless there’s some action the W3C could/should be taking that this issue is blocked on, I suggest we close it and that any issues for further actions that need to be taken by WHATWG editors (or anybody else) be raised as issues elsewhere.

Ideally we want to have zero issues in this repo — as a quantitative indication that there aren’t any specific WHATWG-W3C coordination issues that are currently in need of resolving

**Comment by sideshowbarker** [comment]
To get the IANA Link Relations registry updated, I’ve raised the following issues:

- https://github.com/protocol-registries/link-relations/issues/53
- https://github.com/protocol-registries/link-relations/issues/54
- https://github.com/protocol-registries/link-relations/issues/55
- https://github.com/protocol-registries/link-relations/issues/56

And while raising those, it occurred to me that those issues on their own should be sufficient to the degree that there’s actually no need to raise a separate issue for this in the HTML issue tracker.

Given all that, it seems like we can now actually close this issue, so I’m going ahead and doing that. But @annevk if you still see any reason we need to keep an open issue for this here, then please do feel free to go ahead and re-open it.

**Comment by sideshowbarker** [comment]
> IIUC, the IANA registrations are for Link headers, but only preconnect (and preload) is supported in Link headers. 

Per https://github.com/protocol-registries/link-relations/issues/53#issuecomment-1621209788 from mnot, that’s apparently not the case:
> this registry is not just for the Link HTTP header field; it's for all link relations

So I’ve opened new issues for `prefetch` and `dns-prefetch`, asking for the **Reference** field to be updated:

- https://github.com/protocol-registries/link-relations/issues/57
- https://github.com/protocol-registries/link-relations/issues/58

Given https://github.com/w3c/whatwg-coord/issues/6#issuecomment-1468316328:
> I think we should keep prerender there, it's still implemented inside chrome.

…I’ve opened an issue for `prerender`:

- https://github.com/protocol-registries/link-relations/issues/59

…with the following content:

> 1. Add a note to the **Notes** field, with this content:
>
>    > This relation is not part of any current standard, though support for it is implemented in Blink (the engine used by Google Chrome and other Chromium-based browsers).
>
> 2. Completely remove the contents of the **Reference** field. (The current contents are “[[Resource Hints](https://www.w3.org/TR/resource-hints/)]”, but https://www.w3.org/TR/resource-hints/ now redirects to https://html.spec.whatwg.org/#linkTypes, which now does not contain any definition for the `prerender` link relation.
